custom/enikshay/two_b_release_1/management/commands/enikshay_2b_case_properties.py

```diff
@@ -0,0 +1,496 @@
+                              CASE_TYPE_TRAIL: 'trails'}
+            episodes_to_person = {}
+            for case in self.accessor.get_reverse_indexed_cases(referrals_and_occurrences_to_person.keys()):
+                bucket = type_to_bucket.get(case.type, None)
+                if bucket:
+                    for index in case.indices:
+                        person_id = referrals_and_occurrences_to_person.get(index.referenced_id)
+                        if person_id:
+                            getattr(all_persons[person_id], bucket).append(case)
+                            if case.type == CASE_TYPE_EPISODE:
+                                episodes_to_person[case.case_id] = person_id
+                            break
+
+            for case in self.accessor.get_reverse_indexed_cases(episodes_to_person.keys()):
+                if case.type == CASE_TYPE_DRTB_HIV_REFERRAL:
+                    for index in case.indices:
+                        person_id = episodes_to_person.get(index.referenced_id)
+                        if person_id:
+                            all_persons[person_id].drtb_hiv.append(case)
+                            break
+
+            for person_case_set in all_persons.values():
+                yield person_case_set
+
+    def migrate_person_case_set(self, person):
+        changes = filter(None,
+            [self.migrate_person(person.person, person.occurrences, person.episodes)]
+            + [self.migrate_occurrence(occurrence, person.episodes) for occurrence in person.occurrences]
+            + [self.migrate_episode(episode, person.episodes) for episode in person.episodes]
+            + [self.migrate_test(test, person.person) for test in person.tests]
+            + [self.migrate_referral(referral, person.occurrences) for referral in person.referrals]
+            + [self.migrate_trail(trail, person.occurrences) for trail in person.trails]
+            + [self.open_secondary_owners(drtb_hiv, person.person, person.occurrences)
+               for drtb_hiv in person.drtb_hiv]
+            + [self.close_drtb_hiv(drtb_hiv) for drtb_hiv in person.drtb_hiv]
+        )
+        if self.commit:
+            self.factory.create_or_update_cases(changes)
+
+    @staticmethod
+    def get_open_occurrence(occurrences):
+        occurrences = [case for case in occurrences
+                       if not case.closed and case.type == CASE_TYPE_OCCURRENCE]
+        return occurrences[0] if occurrences else None
+
+    def migrate_person(self, person, occurrences, episodes):
+        self.total_persons += 1
+        occurrence = self.get_open_occurrence(occurrences)
+
+        if occurrence:
+            episodes = [case for case in episodes
+                        if not case.closed and case.type == CASE_TYPE_EPISODE
+                        and any([index.referenced_id == occurrence.case_id for index in case.indices])
+                        and case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
```

Comment:
Yup - I think for updating the person case, we don't really care whether the episode type was confirmed_tb or presumptive_tb.   All we care about is getting the open episode for the open occurrence.